be/src/olap/delta_writer.cpp

```diff
@@ -132,8 +142,10 @@ OLAPStatus DeltaWriter::init() {
 
     _tablet_schema = &(_tablet->tablet_schema());
     _schema = new Schema(*_tablet_schema);
-    _mem_table = new MemTable(_schema, _tablet_schema, _req.slots,
-                              _req.tuple_desc, _tablet->keys_type());
+    _mem_table = std::make_shared<MemTable>(_tablet->tablet_id(), _schema, _tablet_schema, _req.slots, _req.tuple_desc, _tablet->keys_type());
+
+    _flush_queue_idx = _flush_executor->get_queue_idx(_tablet->data_dir()->path_hash());
```

Comment:
Why are you exposing this queue to the outside?

Reply:
Because for now, each DeltaWriter must only push memtables to the same queue.